### DISCUSSION TITLE
SSR621Q  NVR

### DISCUSSION THREAD

**Discussion by hojnikb** [linux-chenxing/linux-chenxing.org#39] [General]
So i'm looking at these cheap ~30$ NVR devices floating around aliexpress. I was thinking this would serve well as a cheap NAS device (given that it has sata support and ethernet). The chipset they're using is apperently SSR621Q. With a bit of googling, this appears to be 621D with 2Gb (so 256MB) of ddr3, instead of 128MB in the 621D.

Anyone toyed with this idea?

**Comment by fifteenhex** [comment]
Do you have a link to the board you're looking at? I can buy one and see how much of our mainline-ish kernel works on it.
Getting SATA working should be possible as the host part is AHCI and we have the source for the SATA PHY driver etc.

**Reply by hojnikb** [reply]
https://www.aliexpress.com/item/1005002498786941.html?spm=a2g0s.9042311.0.0.58e54c4daycac9

This is the exact device i've ordered. I figured if i can't get it to work as a headless server, i can always use it for it's intended purpose.

That would be great, if kernel can be ported to it. I really just need sata, network and some way to install packages.

**Reply by hojnikb** [reply]
https://www.aliexpress.com/item/1005001934166805.html?spm=a2g0o.store_pc_groupList.8148356.2.2c6dd70bjdZ4LU

And here is the bare board. It's not really that much cheaper than the whole deal.

**Reply by fifteenhex** [reply]
~$20 is pretty good with a case!

Board looks really similar to the TP-LINK SSR621D NVR I have:
https://github.com/linux-chenxing/linux-chenxing.org/blob/master/infinity2/tlnvr6108cb/IMG_20200414_172602.jpg

I think with a device tree we should be able to get usb, ethernet working right away. SATA I think just needs the vendor PHY driver ported over...

Did you want to run Debian or similar on this? The flash on it is probably 16MB or less so it'll only be big enough for u-boot and a kernel. I think we'll need SATA working in u-boot to boot Debian.

**Reply by hojnikb** [reply]
Yeah, debian or some other well maintained distro would be great, but even a minimal openwrt like distro would suffice. I really only need SSH, SCP and rysnc to use it as an offsite backup nas. Having samba/nfs would be even better, if that was possible to fit in the flash (if boot from sata wouln't be possible). If uboot sata booting actually works along with debian, that would open other possibilites also.

**Reply by Maksons** [reply]
Can someone approve it have gigabit ethernet ? 
In Aliexpress item description written : Network Interface 10 / 100 / 1000 .
But from board picture I see only Fast Ethernet ( 10/100 ) transformer, am I wrong ?

**Reply by fifteenhex** [reply]
I have a feeling it's a mistake. I haven't seen any SoC from this company with gigabit ethernet yet. I think if it was gigabit there would be an external PHY too and I don't see that.
